SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -370,6 +377,81 @@ void ASpatialDebugger::OnToggleConfigUI()
 	}
 }
 
+void ASpatialDebugger::OnMousePress()
+{
+	if (HitActors.Num() > 0)
+	{
+		TWeakObjectPtr<AActor> SelectedActor = HitActors[HoverIndex];
+
+		if (SelectedActor.IsValid())
+		{
+			if (SelectedActors.Contains(SelectedActor))
+			{
+				// Already selected so deselect
+				SelectedActors.Remove(SelectedActor);
+			}
+			else
+			{
+				// Add selected actor to enable drawing tags
+				SelectedActors.Add(SelectedActor);
+			}
+		}
+	}
+}
+
+void ASpatialDebugger::OnMouseWheelAxis()
+{
+	HoverIndex++;
+	ValidateHoverIndex();
+}
+
+void ASpatialDebugger::ToggleSelectActor(bool bEnable)
+{
+	if (bEnable == bSelectActor)
+	{
+		return;
+	}
+
+	bSelectActor = bEnable;
+	if (bSelectActor)
+	{
+		if (CrosshairTexture != nullptr)
+		{
+			// Hide the mouse cursor as we will draw our own custom crosshair
+			LocalPlayerController->bShowMouseCursor = false;
+			// Sets back the focus to the game viewport - need to hide mouse cursor instantly
+			FSlateApplication::Get().SetAllUserFocusToGameViewport();
+		}
+
+		// Set the object types to query in the raycast based
+		for (TEnumAsByte<ECollisionChannel> ActorTypeToQuery : SelectActorTypesToQuery)
+		{
+			CollisionObjectParams.AddObjectTypesToQuery(ActorTypeToQuery);
+		}
+	}
+	else
+	{
+		// Clear selected actors
+		// Change mouse cursor back to normal - TODO check if it was true before
```

Comment:
Store this value instead?

Reply:
I removed the TODO comment as you can only start / stop selecting actors when the Config UI window is open which sets the mouse cursor to true.